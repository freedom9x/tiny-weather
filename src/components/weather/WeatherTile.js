import React from 'react'
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'

const propTypes = {
  statusIcon: PropTypes.string,
  dayName: PropTypes.string,
  min: PropTypes.number,
  max: PropTypes.number,
  isFetching: PropTypes.bool,
  unit: PropTypes.string,
}

const Weather = styled.div`
    display: flex;
    padding: 5px;
    height: 160px;
    width: 160px;
    background: lightblue;
    margin: 5px;
    color: grey;
    border-radius: 5px;
    font-weight: 600;
    
`

const DayName = styled.div`
  text-align: center;
  font-size: 16px;
`

const TemperatureTree = styled.div`
    padding: 10px;
    flex-direction: column;
    display: flex;
    justify-content: center;
    align-items: center;
`

const DayInfo = styled.div`
    flex-direction: column;
    padding: 10px;
    flex: 1;
    display: flex;
`

const Temperature = styled.div`

`
const Tree = styled.div`
  width: 6px;
  border-radius: 20px;
  background: linear-gradient(to top, #95ca3d, #bf2b1c);
  height: 100%;
  ${({ isCool }) => (isCool ? css` background: linear-gradient(to top,#2da3e4,#c8d018);` : '')
}
`
const WeatherStatusIcon = styled.div`
  background: url(${(props) => props.iconPath}) center no-repeat;
  flex: 1;
`

const SkeletonWeather = styled(Weather)`
  background: linear-gradient(to right,#ffffff, #b7b7b7);
  background-size: 400% 400%;
  animation: gradient 0.5s ease infinite;
`

const WeatherTile = ({
  statusIcon, dayName, min, max, isFetching, unit,
}) => {
  if (isFetching) {
    return (<SkeletonWeather />)
  }
  return (
    <Weather>
      <DayInfo>
        <DayName> {dayName} </DayName>
        <WeatherStatusIcon iconPath={statusIcon}/>
      </DayInfo>
      <TemperatureTree>
        <Temperature> {`${max.toFixed(0)}${String.fromCharCode('0176')}${unit}`}</Temperature>
        <Tree isCool={max < 30}/>
        <Temperature>{`${min.toFixed(0)}${String.fromCharCode('0176')}${unit}`}</Temperature>
      </TemperatureTree>
    </Weather>
  )
}

WeatherTile.propTypes = propTypes
export default WeatherTile
