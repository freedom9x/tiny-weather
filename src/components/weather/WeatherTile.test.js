import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'


import WeatherTile from './WeatherTile'

// import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('WeatherTile ', () => {
  it('render skeleton style correct', () => {
    const tree = mount(<WeatherTile isFetching={true} />)
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('render normal style correct', () => {
    const tree = mount(
      <WeatherTile
        isFetching={false}
        statusIcon='icon.svg'
        dayName='2020-05-25'
        min={10.21}
        max= {21.10}
        unit='C'
        />,
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
