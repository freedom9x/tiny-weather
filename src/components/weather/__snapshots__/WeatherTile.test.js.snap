// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherTile  render normal style correct 1`] = `
<WeatherTile
  dayName="2020-05-25"
  isFetching={false}
  max={21.1}
  min={10.21}
  statusIcon="icon.svg"
  unit="C"
>
  <styled.div>
    <div
      className="sc-AxjAm iqmdHn"
    >
      <styled.div>
        <div
          className="sc-AxhCb gMlMqi"
        >
          <styled.div>
            <div
              className="sc-AxirZ hEbYLu"
            >
               
              2020-05-25
               
            </div>
          </styled.div>
          <styled.div
            iconPath="icon.svg"
          >
            <div
              className="sc-AxheI hSEUAb"
            />
          </styled.div>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-AxiKw gpevwn"
        >
          <styled.div>
            <div
              className="sc-AxhUy dNxuhj"
            >
               
              21°C
            </div>
          </styled.div>
          <styled.div
            isCool={true}
          >
            <div
              className="sc-AxgMl fpklkb"
            />
          </styled.div>
          <styled.div>
            <div
              className="sc-AxhUy dNxuhj"
            >
              10°C
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</WeatherTile>
`;

exports[`WeatherTile  render skeleton style correct 1`] = `
<WeatherTile
  isFetching={true}
>
  <Styled(styled.div)>
    <div
      className="sc-AxjAm sc-Axmtr cQwmQX"
    />
  </Styled(styled.div)>
</WeatherTile>
`;
